### ISSUE TITLE
Weird collisions with ElevationGrid if x and y dimensions diff is 1

### ISSUE THREAD

**Issue by Inngress** [cyberbotics/webots#6363]
**Describe the Bug**
While using an ElevationGrid as a boundingObject in a Shape, if its x and y dimensions have a diff of 1 (e.g. `xDimension 3` and `yDimension 2`), the collisions with any object don't match with what is rendered (and not with what is displayed by the boundingObject optional view neither). At some places the object goes through the floor as if there was nothing and at others it gets stuck on invisible terrain. 
![cube_fell_trough_floor](https://github.com/cyberbotics/webots/assets/74958967/2320ba41-2f29-4753-8755-0bb9e7b0eb16)
![cube_floating_in_the_air](https://github.com/cyberbotics/webots/assets/74958967/e8c5b827-811a-4f72-9957-33a6733828b3)
Also repeating the same tests tend not to give same results.

I tested out on a large scale of points with a map that just had a quarter of it elevated and the rest at a null or negative height and it looked like the boundingObject was kind of mirrored on `y` but no way to be sure since it is rendered at it's intended place. Plus the mirrored bounding object is not well defined, it is not a perfect mirror, there are strange holes and unclear boundaries everywhere while trying to make a robot drive on it on example.

TLDR: real collision is often at the opposite of where it is supposed to be.

Don't know if this could be related to  #4169

**Steps to Reproduce**
1. Download [uneven_terrain_test.zip](https://github.com/cyberbotics/webots/files/12418797/uneven_terrain_test.zip) world and load it with Webots 
3. Move the box around

**Expected behavior**
boundingObject working as intended for any set of dimensions

**System**
 - Operating System: Linux Ubuntu 22.04
 - Graphics Card: NVIDIA GeForce RTX 3070 Ti laptop 8 GB
 - Webots 2023b

**Comment by ShuffleWire** [comment]
I've tried your world and didn't seems to be able to reproduced it.

Your images seems to show that the plane is tilted, when your world you provided is not, is that correct ?
If so, could you provide a precise world, with **all the object** placed exactly in the configuration that trigger the bug, **including the camera position**, so that we can both record a video showing (or not) the bug, and compare apple to apple.

**Comment by Inngress** [comment]
Well, that's kinda awkward... In the world I sent I forgot to uncomment the elevation maps heights... So the floor is indeed flat. My bad.

[Here](https://github.com/cyberbotics/webots/files/12560755/uneven_terrain_test.zip) is a corrected version with exact camera and all. 

I also join gifs to illustrate what I have with the box at different $x$ values
$x=12$
![uneven_terrain_test12](https://github.com/cyberbotics/webots/assets/74958967/bb311d44-7f95-446b-9d1d-790001ee0ab2)
$x=7$
![uneven_terrain_test7](https://github.com/cyberbotics/webots/assets/74958967/e7338f85-0b9f-48f7-9341-acaecf7d8025)
$x=3$
![uneven_terrain_test3](https://github.com/cyberbotics/webots/assets/74958967/b8c7f83c-95fe-49e9-bb3e-364c9c076c5c)


Also I noticed that it happens also with more than 1 dimension diff. Actually maybe any dimension difference.

**Comment by ShuffleWire** [comment]
might be linked to https://github.com/cyberbotics/webots/issues/4169
